### ISSUE TITLE
add-constant reports problem with string even if it should ignore

### ISSUE THREAD

**Issue by heynemann** [mgechev/revive#663]
**Describe the bug**
Even if string is in the `allowStrs` list, once it appears more than `maxLitCount` I get an error. Not sure if this is the desired effect, but had to disable if it is.

**To Reproduce**
1. I updated revive `go get -u github.com/mgechev/revive`
2. I run it with the following flags & configuration file:

```shell
# flags

revive ...
```

```toml
[rule.add-constant]
  arguments = [{maxLitCount = "3",allowStrs ="\"\"",allowInts="0,1,2",allowFloats="0.0,0.,1.0,1.,2.0,2."}]
```

**Expected behavior**
Even if `""` shows 100 times, it should be ignored as it is allowed. Same as `0, 1, 2` and the floats.

**Comment by chavacava** [comment]
Hi @heynemann, thanks for filling the issue.
I was unable to reproduce the problem.
For the intput file `deleteme-663.go`
```go
package main

func foo() {
	print("", 1, 0.0)
	print("", 1, 0.0)
	print("", 1, 0.0)
	print("", 2, 1.0)
	print("", 2, 1.0)
	print("", 2, 1.0)
}
```
With the configuration `deleteme-663.toml`
```toml
ignoreGeneratedHeader = false
severity = "warning"
confidence = 0.8
errorCode = 0
warningCode = 0


[rule.add-constant]
  arguments = [{maxLitCount = "1",allowStrs ="\"\"",allowInts="0,2",allowFloats="0.0,0.,1.0,1.,2.0,2."}]
```

And running `revive` with
```bash
./revive -config deleteme-663.toml deleteme-663.go
```
I obtain the expected output:
```
deleteme-663.go:4:12: avoid magic numbers like '1', create a named constant for it
deleteme-663.go:5:12: avoid magic numbers like '1', create a named constant for it
deleteme-663.go:6:12: avoid magic numbers like '1', create a named constant for it
```

The sandbox I've used to try to reproduce is available in the branch https://github.com/mgechev/revive/tree/reproduce/663
